mesh/mesh.go

```diff
@@ -92,18 +130,51 @@ func (m *Mesh) AddLayer(layer *Layer) error {
 		m.Debug("can't add layer", layer.Index(), " missing previous layers")
 		return errors.New("can't add layer missing previous layers")
 	}
-
+	atomic.StoreUint32(&m.lastSeenLayer, uint32(layer.Index()))
```

Comment:
this basically overwrites lastSeenLayer which could be higher

Reply:
there is a check above that validates that this layer is exactly +1 from previous recevied